storage2/src/storage.rs

```diff
@@ -29,16 +64,129 @@ impl Storage {
     }
 
     // TODO: this probably can't be 'static long-term
-    pub async fn apply(&'static mut self, state: State) {
-        // TODO: 1. write the index tables and JMT to RocksDB
-        // 2. update the snapshot
-        // TODO: set jmt_version correctly
-        let jmt_version = 0;
-        let snapshot = self.db.snapshot();
-        // Obtain the write-lock for the latest snapshot, and replace it with the new snapshot.
-        let mut guard = self.latest_snapshot.write();
-        *guard = Arc::new(Snapshot::new(snapshot, jmt_version));
-        // Drop the write-lock (this will happen implicitly anyways, but it's good to be explicit).
-        drop(guard);
+    pub async fn apply(&'static mut self, state: State, nct: &Tree) -> Result<()> {
```

Comment:
I think it would be more extensible for the future if this method only took a `State`, not a `Tree`, so that we're not special-casing the NCT state.  Instead, we would have methods on `State`/`StateTransaction` for writing key/value pairs that aren't part of the consensus state.

Reply:
Yeah, that makes sense to me. Probably the `State` should have something like a `HashMap<&'static str, HashMap<String, Option<Vec<u8>>>>>` to track changes to column families (named by `&'static str`) that are not the JMT?